core/src/main/java/org/infinispan/factories/AbstractNamedCacheComponentFactory.java

```diff
@@ -13,7 +13,7 @@
  * @since 4.0
  */
 @Scope(Scopes.NAMED_CACHE)
```

Comment:
A factory is in the same scope as the component it constructs, because it usually needs to inject other components from the same scope. So we have both `AbstractNamedCacheComponentFactory` for cache components and `AbstractComponentFactory` for global components.